rockchip: add support for Radxa ROCK Pi E v3.0

RadxaNaoki opened the pull request:
Radxa ROCK Pi E v3.0 is a compact networking SBC[1] using the Rockchip RK3328 SoC.

Hardware
--------
- Rockchip RK3328 SoC
- Quad A53 CPU
- 512MB/1GB/2GB DDR4 RAM
- 4/8/16/32GB eMMC
- Micro SD Card slot
- WiFi 4 and BT 4, or WiFi 5 and BT 5 (not supported yet)
- 1x 1000M Ethernet with PoE support (additional PoE HAT required)
- 1x 100M Ethernet
- 1x USB 3.0 Type-A port (Host)
- 1x 4-ring 3.5mm headphone jack
- 40 Pin GPIO header

[1] https://radxa.com/products/rockpi/pie

Installation
------------
Uncompress the OpenWrt sysupgrade and write it to a micro SD card or internal eMMC using dd.

RadxaNaoki commented:
oh, I found mistake... give me a sec ;)

RadxaNaoki commented:
@hauke could you review this too?

hauke commented on target/linux/rockchip/image/armv8.mk:
```diff
@@ -128,9 +128,20 @@ define Device/radxa_rock-pi-e
   DEVICE_VENDOR := Radxa
   DEVICE_MODEL := ROCK Pi E
   SOC := rk3328
```
Is this needed here too?
`UBOOT_DEVICE_NAME := rock-pi-e-rk3328`

hauke replied:
ah, you use the default setting here.

hauke commented on target/linux/rockchip/image/armv8.mk:
```diff
@@ -128,9 +128,20 @@ define Device/radxa_rock-pi-e
   DEVICE_VENDOR := Radxa
   DEVICE_MODEL := ROCK Pi E
   SOC := rk3328
+  DEVICE_PACKAGES := kmod-usb-net-cdc-ncm kmod-usb-net-rndis
```
Please move the changes for `Device/radxa_rock-pi-e` into a separate commit.

RadxaNaoki committed:
rockchip: add support for Radxa ROCK Pi E v3.0

Radxa ROCK Pi E v3.0 is a compact networking SBC[1] using the Rockchip
RK3328 SoC.

Hardware
--------
- Rockchip RK3328 SoC
- Quad A53 CPU
- 512MB/1GB/2GB DDR4 RAM
- 4/8/16/32GB eMMC
- Micro SD Card slot
- WiFi 4 and BT 4, or WiFi 5 and BT 5 (not supported yet)
- 1x 1000M Ethernet with PoE support (additional PoE HAT required)
- 1x 100M Ethernet
- 1x USB 3.0 Type-A port (Host)
- 1x 4-ring 3.5mm headphone jack
- 40 Pin GPIO header

[1] https://radxa.com/products/rockpi/pie

Installation
------------
Uncompress the OpenWrt sysupgrade and write it to a micro SD card or
internal eMMC using dd.

Signed-off-by: FUKAUMI Naoki <naoki@radxa.com>
Link: https://github.com/openwrt/openwrt/pull/15984
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

RadxaNaoki commented:
I don't have ROCK Pi E v1.x. I cannot test any change on real device, so I dropped changes for ROCK Pi E (v1.x).

about UBOOT_DEVICE_NAME, default value should be fine, I dropped it.

1715173329 commented on package/boot/uboot-rockchip/patches/000-rockchip-add-support-for-Radxa-ROCK-Pi-E-v3.0.patch:
```diff
@@ -0,0 +1,238 @@
+ create mode 100644 arch/arm/dts/rk3328-rock-pi-e-v3.dts
+ create mode 100644 configs/rock-pi-e-v3-rk3328_defconfig
+
+diff --git a/arch/arm/dts/rk3328-rock-pi-e-u-boot.dtsi b/arch/arm/dts/rk3328-rock-pi-e-base-u-boot.dtsi
```
please refresh this patch with `make package/uboot-rockchip/{clean,refresh} V=s`

RadxaNaoki replied:
done
